module/utils/dice-tools.mjs

```diff
@@ -0,0 +1,14 @@
+/**
+ *
+ * Helper function to extract the number of d10 dice from a given value
+ *
+ * @param {String} value
+ * @returns integer
+ */
+export async function metaExtractNumberOfDice(value) {
+	const match = value.match(/^(\d+)d10$/);
```

Comment:
You have a problem. You decide to solve it with Regular Expressions.

... Now you have two problems!